### PR TITLE
Feature/migrate identity server4 to open iddict

### PR DESCRIPTION
**Pull request by KONKdiogo-ferraz**
.Removed IdentityServer4 dependencies.
.Updated Devices Service to .NET 7
.Added OpenIddict identity server.
.Updated dependency packages to the latest version.
.Disable token storage in the database
.Removed the OpenIddict's Quartz integration.
.Updated the token's claims, scopes and permissions.

### CODE REVIEW

**Comment by tnotheis** [comment]
I just looked at the ClientSecret problem. And I saw that there is a method called ValidateClientSecretAsync(...) on the OpenIddictApplicationManager. You can pass the OpenIddictEntityFrameworkCoreApplication and the secret and it returns true or false.

Sounds fitting. 🙂

File: Devices.Application/Devices/Queries/ListDevices/Handler.cs [hunk 1]
```diff
@@ -28,11 +28,12 @@ public async Task<ListDevicesResponse> Handle(ListDevicesQuery request, Cancella
         var query = _dbContext
             .SetReadOnly<Device>()
             .NotDeleted()
+            .IncludeUser()
```
Comment (tnotheis) [inline Devices.Application/Devices/Queries/ListDevices/Handler.cs:31]: I have no idea why this never failed in the past, but when mapping the Device to a DeviceDTO in line 39, the `User` property is accessed, which is `null` when this line is missing.

File: Devices.API/Extensions/IServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -178,10 +124,65 @@ public static void AddCustomIdentity(this IServiceCollection services, IHostEnvi
                 }
             })
             .AddEntityFrameworkStores<ApplicationDbContext>()
+            .AddDefaultTokenProviders()
```
[the remark this answers is not in the document]
Comment (KONKdiogo-ferraz) [reply Devices.API/Extensions/IServiceCollectionExtensions.cs:126]: No, its not required. I guess if we don't plan on having any of the functionalities mentioned above we can safely remove this method call.

[the remark this answers is not in the document]
Comment (tnotheis) [reply Devices.API/Extensions/IServiceCollectionExtensions.cs]: Ok, I did remove it.

**Review by tnotheis — APPROVED** [review]

**Pull request merged**